### ISSUE TITLE
perform_job test helper converts unexpected result to nil

### ISSUE THREAD

**Issue by beerlington** [sorentwo/oban#590]
### Environment

oban 2.9.2
oban_pro 0.9.2
oban_web 2.7.4
Postgres 13.3
Erlang/OTP 24 [erts-12.0.3] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [jit]
Elixir 1.12.2 (compiled with Erlang/OTP 24)

### Current Behavior

Given a simple worker like:

```ex
defmodule MyApp.Worker do
  use Oban.Worker

  @impl Oban.Worker
  def perform(_) do
    Ecto.Multi.new()
    |> Ecto.Multi.run(:test, fn _, _ -> {:error, :test} end)
    |> MyApp.Repo.transaction()
  end
end
```

and test:

```ex
defmodule MyApp.WorkerTest do
  use MyApp.ModelCase
  use Oban.Testing, repo: MyApp.Repo

  describe "perform/1" do
    test "handles Ecto.Multi failure" do
      {:error, _, _, _} = perform_job(MyApp.Worker, %{})
    end
  end
end
```

The `perform_job` test helper converts the real multi result `{:error, :test, :test, %{}}` to `nil`:

```
     Expected result to be one of
     
       - `:ok`
       - `:discard`
       - `{:ok, value}`
       - `{:error, reason}`
       - `{:discard, reason}`
       - `{:snooze, duration}
     
     Instead received:
     
     nil
```
### Expected Behavior

I would expect the multi result to remain intact and not be converted to `nil`.

In production it would be nice if Oban also considered this an error and retried the job rather than considering it a success.

**Comment by sorentwo** [comment]
> I would expect the multi result to remain intact and not be converted to nil.

This was fixed with 3ba18ec, but isn't released yet.

> In production it would be nice if Oban also considered this an error and retried the job rather than considering it a success.

Agreed. That would be a breaking change and would require a bump to Oban 3.X, though. Maybe one day!

**Comment by sorentwo** [comment]
Released in v2.10.0 👍
